doc/visual-programming/source/widgets/model/stochasticgradient.rst

```diff
@@ -0,0 +1,103 @@
+Minimize an objective function using a stochastic approximation of gradient descent.
+
+Signals
+-------
+
+**Inputs**:
+
+-  **Data**
+
+   A data set.
+
+-  **Preprocessor**
+
+   Preprocessing method(s)
+
+**Outputs**:
+
+-  **Learner**
+
+   A SGD regression learning algorithm with settings as specified in the dialog.
+
+-  **Model**
+
+   A trained model. Output signal sent only if input *Data* is present.
+
+Description
+-----------
+
+The **Stochastic Gradient Descent** widget uses `stochastic gradient descent <https://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_ that minimizes a chosen loss function with a linear function. The algorithm approximates a true gradient by considering one sample at a time, and simultaneously updates the model based on the gradient of the loss function. For regression, it returns predictors as minimizers of the sum, i.e. M-estimators, and is especially useful for large-scale and sparse data sets.
+
+.. figure:: images/StochasticGradientDescent-stamped.png
+   :scale: 50 %
+
+1. Specify the name of the model. The default name is "SGD".
+2. Algorithm parameters. Classification loss function:
+
+   -  `Hinge <>`_ ()
+   -  `Logistic Regression <>`_ ()
+   -  `Modified Huber <>`_ ()
+   -  `Squared Hinge <>`_ ()
+   -  `Perceptron <>`_ ()
+   -  `Squared Loss <https://en.wikipedia.org/wiki/Mean_squared_error#Regression>`_
+      (fitted to ordinary least-squares)
+   -  `Huber <https://en.wikipedia.org/wiki/Huber_loss>`_ (switches to
+      linear loss beyond ε)
+   -  `Epsilon insensitive <http://kernelsvm.tripod.com/>`_ (ignores
+      errors within ε, linear beyond it)
+   -  *Squared epsilon insensitive* (loss is squared beyond ε-region).
+
+   Regression loss function:
+
+   -  `Squared Loss <https://en.wikipedia.org/wiki/Mean_squared_error#Regression>`_
+      (fitted to ordinary least-squares)
+   -  `Huber <https://en.wikipedia.org/wiki/Huber_loss>`_ (switches to
+      linear loss beyond ε)
+   -  `Epsilon insensitive <http://kernelsvm.tripod.com/>`_ (ignores
+      errors within ε, linear beyond it)
+   -  *Squared epsilon insensitive* (loss is squared beyond ε-region).
+
+3. Regularization norms to prevent overfitting:
+
+   -  None.
+   -  `Lasso (L1) <https://en.wikipedia.org/wiki/Taxicab_geometry>`_ (L1,
+      leading to sparse solutions)
+   -  `Ridge (L2) <https://en.wikipedia.org/wiki/Norm_(mathematics)#p-norm>`_
+      (L2, standard regularizer)
+   -  `Elastic net <https://en.wikipedia.org/wiki/Elastic_net_regularization>`_
+      (mixing both penalty norms).
+
+   Regularization strength defines how... and mixing parameter how ....
+
+4. Learning parameters.
+
+   - Learning rate:
+
+      - *Constant*: learning rate stays the same through all epochs (passes)
+      - *Optimal*: 
+      - `Inverse scaling <http://users.ics.aalto.fi/jhollmen/dippa/node22.html>`_: earning rate is inversely related to the number of iterations
+
+   - Initial learning rate:
+   - Inverse scaling exponent: 
```

Comment:
The initial learning rate should be fairly self explanatory. The inverse scaling factor seems to be some kind of learning rate decay, which just means that at first we want to make larger steps towards the minimum of the loss function, but later, we expect to be closer to the minimum so we want to make smaller steps.

I'm not really sure about this, because the scikit learn wiki is useless here but by looking at the formula, I don't really know what else it could be.